src/search/merge_and_shrink/shrink_strategy.h

```diff
@@ -6,14 +6,18 @@
 #include <string>
 #include <vector>
 
+namespace utils {
+class LogProxy;
+}
+
 namespace merge_and_shrink {
 class Distances;
 class TransitionSystem;
 
 class ShrinkStrategy {
```

Comment:
From the previous modifications, why did you not add the log options for the parser for the ShrinkStrategies?

Reply:
I think because with the exception of a single line in the bucket-based strategy, shrink strategies don't log anything. Every component of M&S prints their options at the start, but I'm not sure if that justifies adding their logger. Maybe it actually does.

Reply:
For the strategy which logs, is that logging required or could be skip it?

Reply:
It always does it as it is some kind of warning. 

I had another look at the thing and I think that not giving shrink strategies their own logger is good because they don't pre-compute anything. They are basically stateless objects, which means whoever wants to use them should decide from the outside if they want to log or not. The same is not true for the merge strategies: here, only the *factories* have a logger because they might do some pre-computation and log there. But the merge strategies themselves don't have a log and do not log at all. This also very nicely aligns with our decision to not pass or have loggers in merge selectors or merge scoring functions, which are all stateless objects that can be used in different contexts.

Reply:
ok